x-pack/legacy/plugins/security/server/routes/api/v1/authenticate.js

```diff
@@ -109,18 +109,26 @@ export function initAuthenticateApi({ authc: { login, logout }, config }, server
           provider: 'oidc',
           // Checks if the request object represents an HTTP request regarding authentication with OpenID Connect.
           // This can be
+          //  - An HTTP GET request with a query parameter named `redirectURI` that includes URL fragment OpenID Connect
+          //    Provider sent during implicit authentication flow to the Kibana own proxy page that extracted that URL
+          //    fragment and put it into `redirectURI` query string parameter
+          //    to this endpoint
           //  - An HTTP GET request with a query parameter named `iss` as part of a 3rd party initiated authentication
           //  - An HTTP POST request with a parameter named `iss` as part of a 3rd party initiated authentication
           //  - An HTTP GET request with a query parameter named `code` as the response to a successful authentication from
           //    an OpenID Connect Provider
           //  - An HTTP GET request with a query parameter named `error` as the response to a failed authentication from
           //    an OpenID Connect Provider
-          value: {
-            code: request.query && request.query.code,
+          value: request.query && request.query.redirectURI ? {
```

Comment:
__note:__ actually, it probably makes sense to have a dedicated endpoint for implicit flow, haven't thought thoroughly about this.

Reply:
I'll defer to your opinion on this one. I don't think the current implementation is un-readable, but if you think it makes it easier to have a separate end-point, that seems reasonable also.

Reply:
> I don't think the current implementation is un-readable

Great, I don't have strong opinion on that. Let's keep it as is for now then and split whenever we feel we really need to split.